### PR TITLE
bsunanda:Run2-alca60 Have some non-trivial parameters for SiPM hardcoded conditions

### PR DESCRIPTION
**Pull request by bsunanda**
Change default hardcoded conditions for SiPM parameters
SiPMCharacteristics contain for each type of SiPM, # of pixels, parameters for non-linearity correction function (at the moment a second order polynomial), cross talk parameter and two auxiliary words. These numbers are taken from data sheets and some measurements.
SiPMParameter contains for each DetID, SiPM type, PE to FC factor, dark current - again coming from some measurements

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CalibFormats/HcalObjects/interface/HcalSiPMType.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#ifndef CALIBFORMATS_HCALOBJECTS_HCALSIPMTYPER_H
+#define CALIBFORMATS_HCALOBJECTS_HCALSIPMTYPER_H 1
+
+enum HcalSiPMType {HcalHOZercotek=0, HcalHOHamamatsu=1, HcalHEHamamatsu1=2,
```
Comment (kpedro88) [inline CalibFormats/HcalObjects/interface/HcalSiPMType.h]: typo: Zercotek -> Zecotek

File: CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc [hunk 2]
```diff
@@ -784,11 +785,24 @@ void HcalDbHardcode::makeHardcodeFrontEndMap(HcalFrontEndMap& emap, const std::v
...
+    } else if (fId.subdetId() == HcalOuter) {
+      return HcalSiPMParameter(fId.rawId(), HcalHOHamamatsu, 4.0, 0.055, 0, 0);
+    }
+  } 
   return HcalSiPMParameter(fId.rawId(), 0, 0, 0, 0, 0);
 }
 
 void HcalDbHardcode::makeHardcodeSiPMCharacteristics (HcalSiPMCharacteristics& sipm) {
-  sipm.loadObject(0,0,0,0,0,0,0,0);
+  sipm.loadObject(HcalHOZercotek,36000,1,0,0,0.32,0,0);
```
Comment (kpedro88) [inline CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc]: typo: Zercotek -> Zecotek

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
+1

File: CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc [hunk 3]
```diff
@@ -784,11 +785,24 @@ void HcalDbHardcode::makeHardcodeFrontEndMap(HcalFrontEndMap& emap, const std::v
...
+    } else if (fId.subdetId() == HcalOuter) {
+      return HcalSiPMParameter(fId.rawId(), HcalHOHamamatsu, 4.0, 0.055, 0, 0);
+    }
+  } 
   return HcalSiPMParameter(fId.rawId(), 0, 0, 0, 0, 0);
 }
 
 void HcalDbHardcode::makeHardcodeSiPMCharacteristics (HcalSiPMCharacteristics& sipm) {
-  sipm.loadObject(0,0,0,0,0,0,0,0);
+  sipm.loadObject(HcalHOZecotek,36000,1,0,0,0.32,0,0);
```
Comment (davidlange6) [inline CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc]: @bsunanda - please add a few lines of documentation on what these numbers are/where they come from. Thanks!

**Comment by bsunanda** [comment]
@davidlange6 I have added a few lines in the comments. Hope this is adequate - or should we try to add this in the code - because these are obvious in the CondFormats/HcalObjects where the classes are defined

**Comment by bsunanda** [comment]
@ggovi Please approve this PR

**Comment by kpedro88** [comment]
@bsunanda did you mean to push a commit with the new comments?

**Comment by bsunanda** [comment]
@kpedro88 @davidlange6 We have to make some changes to HcalTPParameter - we shall add the comment in the code at that time. For this PR I have added the doc lines in the comment block. Hope this is acceptable to David

**Pull request merged**